.grit/patterns/common.grit

```diff
@@ -45,55 +45,38 @@ predicate IsUsed($name) = {
 predicate RemoveImportIfUnused($name) = (! IsUsed($name)) && RemoveImport($name)
 
 predicate IsTypeScript() = $program <: or {
-    File(name = or { r".*tsx?", r".*ts?" })
+    File(name = or { r".*tsx?", r".*ts?" }),
     File(program = contains { TSTypeAnnotation() })
 }
 
 pattern ImportedFrom($from) = $name where or {
-    $name <: semantic within ImportNameFrom($name, $from)
-    $name <: semantic `require($from)`
+    $name <: semantic within ImportNameFrom($name, $from),
+    $name <: semantic `require($from)`,
     IsImported($name, $from)
   }
 
 predicate AddImport($m, $fromArg) =
   $fromArg <: let $from in or {
-    StringLiteral() as $from
+    StringLiteral() as $from,
     $_ where $from = StringLiteral(value = $fromArg)
   } where {
-    ! $m <: $alreadySolved
-    $alreadySolved = or { $alreadySolved, $m }
-    $solvedPairs = or { $solvedPairs, `$m, $from` }
-    or { and { ! $from <: $froms, $froms = or { $froms, $from } }, true }
-  }
-
-predicate AddNamespaceImport($m, $fromArg) =
```

Comment:
Why is removing this safe?

Reply:
copied from prelude in the engine, will add back.